db.js

```diff
@@ -114,49 +121,35 @@ exports.init = function (sbot, dir, config) {
     })
   }
 
-  function del(key, cb) {
+  function del(msgId, cb) {
     const guard = guardAgainstDuplicateLogs('del()')
     if (guard) return cb(guard)
 
-    // FIXME: this doesn't work anymore after changing base index
-    baseIndex.keyToSeq(key, (err, seq) => {
+    jitdb.all(key(msgId), 0, false, true, (err, results) => {
       if (err) return cb(err)
-      if (seq == null) return cb(new Error('seq is null!'))
+      if (results.length === 0) return cb(new Error('seq is null!'))
 
-      log.del(seq, cb)
+      log.del(results[0], cb)
     })
   }
 
   function deleteFeed(feedId, cb) {
     const guard = guardAgainstDuplicateLogs('deleteFeed()')
     if (guard) return cb(guard)
 
-    // FIXME: doesn't work, need test
-    jitdb.onReady(() => {
-      jitdb.query(
-        {
-          type: 'EQUAL',
-          data: {
-            seek: jitdb.seekAuthor,
-            value: feedId,
-            indexType: 'author',
-          },
-        },
-        (err, results) => {
-          push(
-            push.values(results),
-            push.asyncMap((msg, cb) => {
-              del(msg.key, cb)
-            }),
-            push.collect((err) => {
-              if (!err) {
-                delete state.feeds[feedId]
-                baseIndex.removeFeedFromLatest(feedId)
-              }
-              cb(err)
-            })
-          )
-        }
+    jitdb.all(author(feedId), 0, false, true, (err, results) => {
+      push(
+        push.values(results),
+        push.asyncMap((seq, cb) => {
+          log.del(seq, cb)
+        }),
+        push.collect((err) => {
+          if (!err) {
```

Comment:
This seems a bit backwards and made me stop and read it carefully. I think if it were written in a more standard way, would make it easy to skim:

```js
if (err) cb(err)
else {

}
```